---
title: "Hackathon 2022"
author: "Akhil Pampana, Nick Sumpter, and Frank Qiang"
date: "8/5/2022"
output: html_document
---

```{r setup, eval = FALSE}
# Location of code on Cheaha.
setwd("/data/project/ubrite/hackathon2022/staging_area_teams/HeriVar")

# Loading required packages.
require(data.table)
require(ggplot2)
require(tibble)
```

```{bash, eval = FALSE}
# Code for copying over to jetstream2
scp * exouser@149.165.159.63:/media/volume/sdb
# Pass: DANA TEND MIRE GRIN HIVE JILL GERM BALE BACH RID BOYD
```

```{r Importing panel and filtering to super pop, eval = FALSE}
# Reading in phenotype information for 1000G cohort.
data = fread("integrated_call_samples_v3.20130502.ALL.panel")

# Subsetting into each of the four populations of interest (excluding AMR).
eur = data[which(data$super_pop %in% c("EUR")),]
afr = data[which(data$super_pop %in% c("AFR")),]
sas = data[which(data$super_pop %in% c("SAS")),]
eas = data[which(data$super_pop %in% c("EAS")),]

# Randomly sampling 489 samples from the three populations that have more than 489 individuals.
eur_random <- eur[sample(nrow(sas)),]
afr_random <- afr[sample(nrow(sas)),]
eas_random <- eas[sample(nrow(sas)),]

# Combining together into final phenotype file.
random_samples = rbind(eur_random,afr_random,eas_random,sas)
```

```{r tabulate random_samples, eval = FALSE}
# Tabulating gender distribution per population.
ftable(random_samples$super_pop,random_samples$gender)

# Printing the dimensions of the table.
dim(random_samples)

# Writing out the resulting sample IDs to the file Samples_to_input.tsv.
write.table(random_samples$sample,file="/data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Phenotypes/Samples_to_input.tsv",row.names=F,col.names=F,quote=F,sep=",",dec = ".")
```

Bash script for filtering vcf files:

```{bash, eval = FALSE}
#!/bin/bash
#SBATCH --ntasks=1
#SBATCH --mem-per-cpu=14096
#SBATCH --partition=medium
#SBATCH --time=48:00:00
#SBATCH --mail-type=FAIL
#SBATCH --mail-user=apampana@uabmc.edu
#SBATCH --output=/data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/out_%J_%j.log
#SBATCH --error=/data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/err_%J_%j.log

module load BCFtools/1.3.1-goolf-1.7.20
module load tabix/0.2.6-goolf-1.7.20

bcftools view -i 'FILTER="PASS"' -S /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Phenotypes/Samples_to_input.tsv --force-sample /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Dataset/CCDG_14151_B01_GRM_WGS_2020-08-05_chr$SLURM_ARRAY_TASK_ID.filtered.shapeit2-duohmm-phased.vcf.gz  -Ov | bgzip -c > /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/chr$SLURM_ARRAY_TASK_ID.pass_only.vcf.gz

tabix -pvcf /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/chr$SLURM_ARRAY_TASK_ID.pass_only.vcf.gz
```

Converting vcf files to plink files:

```{bash, eval = F}
#!/bin/bash
#SBATCH --ntasks=1
#SBATCH --mem-per-cpu=14096
#SBATCH --partition=medium
#SBATCH --job-name=PLINK_CONVERSION_1000g_HH_2022
#SBATCH --time=48:00:00
#SBATCH --mail-type=FAIL
#SBATCH --mail-user=apampana@uabmc.edu
#SBATCH --output=/data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/PLINK_files/out_%J_%j.log
#SBATCH --error=/data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/PLINK_files/err_%J_%j.log

module load PLINK/2.00-alpha3-x86_64

plink2 --vcf /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/VCFS_subset/chr$SLURM_ARRAY_TASK_ID.pass_only.vcf.gz --make-bed --geno 0.05 --maf 0.01 --out /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/PLINK_files/chr$SLURM_ARRAY_TASK_ID.pass_only_filtered
```

Making plink allele frequency files:

```{bash, eval = F}
for i in {1..22}; do plink2 --bfile chr${i}.pass_only_filtered --freq --out chr${i}.pass_only_filtered_frq; done
```

Analyzing allele frequency files:

```{r, eval = FALSE}
for(i in 1:22){
  assign(paste0("chr", i), read.table(paste0("/data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/PLINK_files", "/chr", i, ".pass_only_filtered_frq.afreq")))
}

out <- c()
for(i in 1:22){
  out <- rbind(out, get(paste0("chr", i)))
}

ggplot(data = out, mapping = aes(x = V5)) + 
  geom_histogram()

ggplot(data = out, mapping = aes(x = V5, color = as.factor(V1))) + 
  geom_freqpoly() +
  labs(x = "Allele Frequency", color = "Chromosome") +
  theme_bw()

ggplot(data = out, mapping = aes(x = V5, color = as.factor(V1))) + 
  geom_freqpoly(stat = "density") +
  labs(x = "Allele Frequency", color = "Chromosome") +
  theme_bw()
```

Making file for pruning array:

```{r, eval = FALSE}
chr <- 1:22

window <- c(250, 500, 1000, 10000)

ld <- c(0.2, 0.4, 0.6, 0.8)

table <- tibble("chr" = rep(chr, each = 16), 
                "window" = rep(rep(window, each = 4), 22), 
                "ld" = rep(rep(ld, 4), 22))

write.table(table,file="/data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Phenotypes/input_for_pruning.tsv",row.names=F,col.names=F,quote=F,sep="\t",dec = ".")
```

Producing variants for pruning:

```{bash, eval = FALSE}
#!/bin/bash
#SBATCH --ntasks=1
#SBATCH --mem-per-cpu=14096
#SBATCH --partition=medium
#SBATCH --job-name=PRUNE_TRESH_HH_2022
#SBATCH --time=48:00:00
#SBATCH --mail-type=FAIL
#SBATCH --mail-user=apampana@uabmc.edu
#SBATCH --output=/data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/PRUNE_VAR/prune_%J_%j.log
#SBATCH --error=/data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/PRUNE_VAR/prune_%J_%j.log

module load PLINK

n=$SLURM_ARRAY_TASK_ID
readarray lines < /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Phenotypes/input_for_pruning.tsv  #### Zero indexed so array job starts with 0
line=${lines[$n]}
chr=$(echo "${line}" | awk -F '\t' '{print $1}')
rsq=$(echo "${line}" | awk -F '\t' '{print $3}')
window=$(echo "${line}" | awk -F '\t' '{print $2}')

echo "${chr}"
echo "${rsq}"
echo "${window}"

plink2 --bfile /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/PLINK_files/chr${chr}.pass_only_filtered  \
        --indep-pairwise ${window} ${rsq}  \
        --out /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/PRUNE_VAR/chr${chr}_${rsq}_${window}
```

Prune plink files:

```{bash, eval = FALSE}
#!/bin/bash
#SBATCH --ntasks=1
#SBATCH --mem-per-cpu=14096
#SBATCH --partition=medium
#SBATCH --job-name=PRUNE_SUBJECT_HH_2022
#SBATCH --time=48:00:00
#SBATCH --mail-type=FAIL
#SBATCH --mail-user=apampana@uabmc.edu
#SBATCH --output=/data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/PRUNE_SUBSET/prune_subset_%J_%j.log
#SBATCH --error=/data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/PRUNE_SUBSET/prune_subset_%J_%j.log
module load PLINK
n=$SLURM_ARRAY_TASK_ID
readarray lines < /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Phenotypes/input_for_pruning_v1.tsv  #### Zero indexed so array job starts with 0
line=${lines[$n]}
chr=$(echo "${line}" | awk -F '\t' '{print $1}')
rsq=$(echo "${line}" | awk -F '\t' '{print $3}')
window=$(echo "${line}" | awk -F '\t' '{print $2}')
echo "${chr}"
echo "${rsq}"
echo "${window}"
plink2 --bfile /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/PLINK_files/chr${chr}.pass_only_filtered  \
        --extract /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/PRUNE_VAR/chr${chr}_${rsq}_${window}.prune.in \
        --set-all-var-ids @:# \
        --make-bed \
        --out /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/PRUNE_SUBSET/PRE_HIGH_LD_REMOVAL/plink/chr${chr}_${rsq}_${window}
```

Making edited highLD.txt file with row numbers:

```{bash, eval = FALSE}
cat HighLD.txt | awk '{print $0, "region"NR}' OFS='\t' > HighLD_v2.txt
```

Subset high LD variants:

```{bash, eval = FALSE}
#!/bin/bash
#SBATCH --ntasks=1
#SBATCH --mem-per-cpu=14096
#SBATCH --partition=medium
#SBATCH --job-name=HIGHLD_HH_2022
#SBATCH --time=48:00:00
#SBATCH --mail-type=FAIL
#SBATCH --mail-user=apampana@uabmc.edu
#SBATCH --output=/data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/PLINK_files/highld_%J_%j.log
#SBATCH --error=/data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/PLINK_files/highld_%J_%j.log
module load PLINK
n=$SLURM_ARRAY_TASK_ID
readarray lines < /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Phenotypes/HighLD_v2.txt  #### Zero indexed so array job starts with 0
line=${lines[$n]}
chr=$(echo "${line}" | awk -F '\t' '{print $1}')
stop=$(echo "${line}" | awk -F '\t' '{print $3}')
start=$(echo "${line}" | awk -F '\t' '{print $2}')
region=$(echo "${line}" | awk -F '\t' '{print $4}')
echo "${chr}"
echo "${start}"
echo "${stop}"
plink2 --bfile /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/PLINK_files/${chr}.pass_only_filtered  \
        --chr ${chr} --from-bp ${start} --to-bp ${stop} \
        --write-snplist \
        --out /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Phenotypes/${chr}_highld_${region}
```

Pruning plink with high LD removed:

```{bash, eval = F}
#!/bin/bash
#SBATCH --ntasks=1
#SBATCH --mem-per-cpu=14096
#SBATCH --partition=medium
#SBATCH --job-name=POST_LD_REMOVAL_HH_2022
#SBATCH --time=48:00:00
#SBATCH --mail-type=FAIL
#SBATCH --mail-user=apampana@uabmc.edu
#SBATCH --output=/data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/PRUNE_SUBSET/POST_HIGH_LD_REMOVAL/logs/HIGH_LD_SUBSET_%J_%j.log
#SBATCH --error=/data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/PRUNE_SUBSET/POST_HIGH_LD_REMOVAL/logs/err_HIGH_LD_SUBSET_%J_%j.log
module load PLINK
n=$SLURM_ARRAY_TASK_ID
readarray lines < /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Phenotypes/input_for_pruning_v1.tsv  #### Zero indexed so array job starts with 0
line=${lines[$n]}
chr=$(echo "${line}" | awk -F '\t' '{print $1}')
rsq=$(echo "${line}" | awk -F '\t' '{print $3}')
window=$(echo "${line}" | awk -F '\t' '{print $2}')
echo "${chr}"
echo "${rsq}"
echo "${window}"
plink2 --bfile /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/PLINK_files/chr${chr}.pass_only_filtered  \
        --extract /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/PRUNE_VAR/chr${chr}_${rsq}_${window}.prune.in \
        --exclude /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Phenotypes/HighLD_Var.tsv \
        --set-all-var-ids @:# \
        --make-bed \
        --out /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/PRUNE_SUBSET/POST_HIGH_LD_REMOVAL/plink/chr${chr}_${rsq}_${window}
#plink2 --bfile /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/PRUNE_SUBSET/PRE_HIGH_LD_REMOVAL/plink/chr${chr}_${rsq}_${window} \
#        --exclude /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Phenotypes/HighLD_Var.tsv \
```


Calculating LD score after removing high LD regions:

```{bash, eval = FALSE}
#!/bin/bash
#SBATCH --ntasks=1
#SBATCH --mem-per-cpu=14096
#SBATCH --partition=medium
#SBATCH --job-name=LDSC_POST_HLD_REM_HH_2022
#SBATCH --time=48:00:00
#SBATCH --mail-type=FAIL
#SBATCH --mail-user=apampana@uabmc.edu
#SBATCH --output=/data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/LD_SCORES/POST_HIGH_LD_REMOVAL/logs/ld_score_%J_%j.log
#SBATCH --error=/data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/LD_SCORES/POST_HIGH_LD_REMOVAL/logs/err_ld_score_%J_%j.log
#module load PLINK
#source /home/nicks95/.bashrc
source /home/apampana/.bashrc
conda activate ldsc
n=$SLURM_ARRAY_TASK_ID
readarray lines < /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Phenotypes/input_for_pruning_v1.tsv  #### Zero indexed so array job starts with 0
line=${lines[$n]}
chr=$(echo "${line}" | awk -F '\t' '{print $1}')
rsq=$(echo "${line}" | awk -F '\t' '{print $3}')
window=$(echo "${line}" | awk -F '\t' '{print $2}')
echo "${chr}"
echo "${rsq}"
echo "${window}"
python /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Tools/ldsc/ldsc.py --bfile /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/PRUNE_SUBSET/POST_HI
GH_LD_REMOVAL/plink/chr${chr}_${rsq}_${window}  --l2 --ld-wind-kb ${window} --out /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/LD_SCORES/POST_HIGH_LD_REMOVAL/sco
res/chr${chr}_${rsq}_${window}
```

Lift over bld positions:

```{bash, eval = FALSE}
#!/bin/bash
#SBATCH --ntasks=1
#SBATCH --mem-per-cpu=14096
#SBATCH --partition=medium
#SBATCH --job-name=liftover2
#SBATCH --time=48:00:00
#SBATCH --mail-type=FAIL
#SBATCH --mail-user=nicks95@uab.edu
#SBATCH --output=/data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/liftover2_%J_%j.log
#SBATCH --error=/data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/liftover2_%J_%j.log
FILE_PATH=/data/project/ubrite/hackathon2022/staging_area_teams/HeriVar
#unzip $FILE_PATH/Results/LDAK/bld.zip # run this once to refresh the bld files
cat $FILE_PATH/Results/LDAK/bld$SLURM_ARRAY_TASK_ID | cut -d ' ' -f 1 | awk '{print "chr"$1,$2-1,$2,"line"NR}' FS=':' OFS='\t' > $FILE_PATH/Results/LDAK/bld$SLURM_ARRAY_TASK_ID.bed
$FILE_PATH/Tools/liftOver $FILE_PATH/Results/LDAK/bld$SLURM_ARRAY_TASK_ID.bed $FILE_PATH/Tools/hg19ToHg38.over.chain $FILE_PATH/Results/LDAK/lifted_bld$SLURM_ARRAY_TASK_ID.bed $FILE_PATH
/Results/LDAK/unlifted_bld$SLURM_ARRAY_TASK_ID.bed
cat $FILE_PATH/Results/LDAK/bld$SLURM_ARRAY_TASK_ID | awk '{print "line"NR,$0}' OFS='-' > $FILE_PATH/Results/LDAK/modified_bld$SLURM_ARRAY_TASK_ID
cat $FILE_PATH/Results/LDAK/lifted_bld$SLURM_ARRAY_TASK_ID.bed | sed 's/chr//g' | awk '{print $4,$1":"$3}' OFS='-' > $FILE_PATH/Results/LDAK/modified_lifted_bld$SLURM_ARRAY_TASK_ID 
join -1 1 -2 1 -t - -o 2.2,1.2 $FILE_PATH/Results/LDAK/modified_bld$SLURM_ARRAY_TASK_ID $FILE_PATH/Results/LDAK/modified_lifted_bld$SLURM_ARRAY_TASK_ID | sed -E 's/-\S+//g' | sort | uniq
 > $FILE_PATH/Results/LDAK/bld$SLURM_ARRAY_TASK_ID
```

LDAK script:

```{bash, eval = FALSE}
#!/bin/bash
#SBATCH --ntasks=1
#SBATCH --mem-per-cpu=14096
#SBATCH --partition=medium
#SBATCH --job-name=LDAK_HH_2022
#SBATCH --time=48:00:00
#SBATCH --mail-type=FAIL
#SBATCH --mail-user=nicks95@uab.edu
#SBATCH --output=/data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/LDAK/out_%J_%j.log
#SBATCH --error=/data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/LDAK/err_%J_%j.log
#n=$SLURM_ARRAY_TASK_ID
#readarray lines < /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Phenotypes/input_for_pruning_v1.tsv  #### Zero indexed so array job starts with 0
#line=${lines[$n]}
#chr=$(echo "${line}" | awk -F '\t' '{print $1}')
#rsq=$(echo "${line}" | awk -F '\t' '{print $3}')
#window=$(echo "${line}" | awk -F '\t' '{print $2}')
chr=22
rsq=0.2
window=250
/data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Tools/LDAK/ldak5.2.linux --cut-weights /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/LDAK/sections${
chr}_${rsq}_${window} --bfile /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/PRUNE_SUBSET/PRE_HIGH_LD_REMOVAL/plink/chr${chr}_${rsq}_${window}
/data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Tools/LDAK/ldak5.2.linux --calc-weights-all /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/LDAK/secti
ons${chr}_${rsq}_${window} --bfile /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/PRUNE_SUBSET/PRE_HIGH_LD_REMOVAL/plink/chr${chr}_${rsq}_${window}
mv /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/LDAK/sections${chr}_${rsq}_${window}/weights.short /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/
Results/LDAK/bld65
/data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Tools/LDAK/ldak5.2.linux --calc-tagging BLD-LDAK --bfile /data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Resul
ts/PRUNE_SUBSET/PRE_HIGH_LD_REMOVAL/plink/chr${chr}_${rsq}_${window} --ignore-weights YES --power -0.25 --window-kb ${window} --annotation-number 65 --annotation-prefix /data/project/ubr
ite/hackathon2022/staging_area_teams/HeriVar/Results/LDAK/bld --allow-multi YES
```

Lifting over variants for urate summary statistics from hg19 to hg38:

```{bash, eval = FALSE}
#!/bin/bash
#SBATCH --ntasks=1
#SBATCH --mem-per-cpu=14096
#SBATCH --partition=medium
#SBATCH --job-name=liftover
#SBATCH --time=48:00:00
#SBATCH --mail-type=FAIL
#SBATCH --mail-user=nicks95@uab.edu
#SBATCH --output=/data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/liftover_%J_%j.log
#SBATCH --error=/data/project/ubrite/hackathon2022/staging_area_teams/HeriVar/Results/liftover_%J_%j.log

FILE_PATH=/data/project/ubrite/hackathon2022/staging_area_teams/HeriVar

cat $FILE_PATH/Phenotypes/urate_chr1_22_LQ_IQ06_mac10_all_741_nstud37_summac400_rsid.txt | tail -n +2 | awk '{print "chr"$1,$2-1,$2,$3}' OFS='\t' > $FILE_PATH/Phenotypes/urate_pos.bed

$FILE_PATH/Tools/liftOver $FILE_PATH/Phenotypes/urate_pos.bed $FILE_PATH/Tools/hg19ToHg38.over.chain $FILE_PATH/Phenotypes/lifted_urate.bed $FILE_PATH/Phenotypes/unlifted.bed
join -1 3 -2 4 -o 2.2,1.2 $FILE_PATH/Phenotypes/urate_chr1_22_LQ_IQ06_mac10_all_741_nstud37_summac400_rsid.txt $FILE_PATH/Phenotypes/lifted_urate.bed | sed -E 's/-\S+//g' > $FILE_PATH/Re
sults/LDAK/bld$SLURM_ARRAY_TASK_ID
```

```{r}
data1 <- read.table("~/chr1.pass_only_filtered_eas_frq.afreq")
data2 <- read.table("~/chr1.pass_only_filtered_afr_frq.afreq")
data3 <- read.table("~/chr1.pass_only_filtered_sas_frq.afreq")
data4 <- read.table("~/chr1.pass_only_filtered_eur_frq.afreq")

data1$ancestry <- "eas"
data2$ancestry <- "afr"
data3$ancestry <- "sas"
data4$ancestry <- "eur"

out <- c()
for(i in 1:4){
  out <- rbind(out, get(paste0("data", i)))
}

ggplot(data = out, mapping = aes(x = V5, color = ancestry)) + 
  geom_freqpoly() +
  labs(x = "Allele Frequency", color = "Ancestry") +
  theme_bw()

ggplot(data = out, mapping = aes(x = V5, color = ancestry)) + 
  geom_density() +
  labs(x = "Allele Frequency", color = "Ancestry") +
  theme_bw()

ggplot(data = out, mapping = aes(x = V5, fill = ancestry)) + 
  geom_density() +
  labs(x = "Allele Frequency", fill = "Ancestry") +
  theme_bw() + 
  facet_wrap(~ancestry)

ggsave(filename = "allelefreq_ancestry.png", dpi = 600, width = 6, height = 4)

ggplot(data = out, mapping = aes(x = V5)) + 
  geom_density() +
  labs(x = "Allele Frequency") +
  theme_bw()

ggsave(filename = "allelefreq.png", dpi = 600, width = 6, height = 4)

post <- read.table("~/chr22_0.2_250.l2.ldscore", header = T)
pre <- read.table("~/chr22_0.2_250.l2.ldscore(1)", header = T)
post$high <- "post"
pre$high <- "pre"

final <- rbind(post, pre)
ggplot(data = final, mapping = aes(x = L2, color = high)) + 
  geom_freqpoly() +
  labs(x = "LD Score", color = "High LD Removed?") +
  theme_bw()

ggplot(data = pre, mapping = aes(x = BP, y = L2)) + 
  geom_point(alpha = 0.5) +
  labs(x = "BP", y = "LD Score") +
  theme_bw()

ggsave("ldsc_22.png", dpi = 600, width = 6, height = 4)
```

